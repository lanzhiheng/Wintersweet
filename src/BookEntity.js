import React, { Component } from 'react';
import './style/BookEntity.css'

class BookEntity extends Component {
  render() {
    return (
      <div className="book-entity">
      <strong>Name</strong><em>活法</em>
      <strong>Author</strong><em>稻盛和夫</em>
      <strong>State</strong><em>Finish</em>
      </div>
    );
  }
}

export default BookEntity;
