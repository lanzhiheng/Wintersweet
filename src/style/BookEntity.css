.book-entity strong{
  color: red;
}
