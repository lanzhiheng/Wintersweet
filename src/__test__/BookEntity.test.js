import React from 'react';
import BookEntity from '../BookEntity';
import renderer from 'react-test-renderer';
const books = require('../data/books-test.json');

test('BookEntity display', () => {
  const component = renderer.create(
    <BookEntity book={books[0]['books'][0]} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // 是否具有某个类
  expect(tree.props.className).toMatch(/\bbook-entity\b/);
});
