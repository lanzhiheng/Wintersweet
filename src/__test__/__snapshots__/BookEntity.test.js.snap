exports[`test BookEntity display 1`] = `
<div
  className="book-entity">
  <strong>
    Name
  </strong>
  <em>
    活法
  </em>
  <strong>
    Author
  </strong>
  <em>
    稻盛和夫
  </em>
  <strong>
    State
  </strong>
  <em>
    Finish
  </em>
</div>
`;
